app/scripts/modules/core/src/pipeline/service/execution.service.ts

```diff
@@ -415,6 +415,13 @@ export class ExecutionService {
         application.executions.data.push(re);
       }
     });
+    application.executions.data.forEach((execution: IExecution) => {
+      if (execution.isActive && application.runningExecutions.data.every((e: IExecution) => e.id !== execution.id)) {
+        this.getExecution(execution.id).then(updatedExecution => {
+          this.updateExecution(application, updatedExecution);
+        });
+      }
+    });
```

Comment:
This should have always been there - if an execution completes, we need to make one last call to get it and merge it in; otherwise, it will sit in a running state until the `executions` data source itself refreshes.